provision/acc_provision/acc_provision.py

```diff
@@ -1491,24 +1511,108 @@ def addMiscConfig(config):
+                postIt(path, data)
+
+            # setup contracts
+            for f in config["aci_config"]["contracts"]:
+                path, data = configurator.make_contract(f)
+                postIt(path, data)
+
+            return "", None
+
+        def postIt(path, data):
+            if args.debug:
+                print("Path: {}".format(path))
+                print("data: {}".format(data))
             try:
                 resp = apic.post(path, data)
                 print("Resp: {}".format(resp.text))
             except Exception as e:
                 err("Error in provisioning %s: %s" % (path, str(e)))
 
+        def getTenantAccount():
+            tn_name = config["aci_config"]["cluster_tenant"]
+            tn_path = "/api/mo/uni/tn-%s.json?query-target=subtree&target-subtree-class=cloudAwsProvider" % (tn_name)
+            resp = apic.get(path=tn_path)
+            resJson = json.loads(resp.content)
+            accountId = resJson["imdata"][0]["cloudAwsProvider"]["attributes"]["accountId"]
+            print(accountId)
+
+        # if underlay ccp doesn't exist, create one
+        underlay_posts = []
+        u_ccp = getUnderlayCCP()
+        if not u_ccp:
+            print("Creating VPC, you will need additional settings for IPI\n")
+            underlay_posts = [configurator.capic_underlay_vrf, configurator.capic_underlay_cloudApp, configurator.capic_underlay_ccp]
+        else:
+            underlay_posts = [underlayCidr, configurator.capic_underlay_cloudApp]
+
+        underlay_posts.append(setupCapicContractsInline)
+
+        postGens = underlay_posts + [configurator.capic_kube_dom, configurator.capic_overlay_vrf, overlayCtx, configurator.capic_overlay_cloudApp, clusterInfo, configurator.capic_kafka_topic, prodAcl, consAcl]
+        for pGen in postGens:
+            path, data = pGen()
+            if not path:  # posted inline
+                continue
+
+            postIt(path, data)
+
         config = addKafkaConfig(config)
         config = addMiscConfig(config)
 
         gen = flavor_opts.get("template_generator", generate_kube_yaml)
         gen(config, output_file, output_tar, operator_cr_output_file)
+        m_cidr = config["net_config"]["machine_cidr"]
+        b_subnet = config["net_config"]["bootstrap_subnet"]
+        n_subnet = config["net_config"]["node_subnet"]
+        p_subnet = config["net_config"]["pod_subnet"].replace(".1/", ".0/")
+        region = config["aci_config"]["vrf"]["region"]
+        boot_subnetID = getSubnetID(b_subnet)
+        node_subnetID = getSubnetID(n_subnet)
+        print("\nOpenshift Info")
+        print("----------------")
+        print("networking:\n  clusterNetwork:\n  - cidr: {}\n    hostPrefix: 23\n  machineCIDR: {}\n  networkType: CiscoAci\n  serviceNetwork:\n  - 172.30.0.0/16\nplatform:\n  aws:\n    region: {}\n    subnets:\n    - {}\n    - {}".format(p_subnet, m_cidr, region, boot_subnetID, node_subnetID))
```

Comment:
Do we need this print statement? Maybe in debug mode?

Reply:
This needs to be copied into the install_config.yaml, so we need this printed until we have a way to automate that edit.